PRESIDECMS-3133 / PRESIDECMS-3134 Added email template stats to Email and Newsletters admin and enabled data export

jamesallenuk opened the pull request.

jamesallenuk committed:
[twgit] Init feature 'feature-PRESIDECMS-3133_Email-Newsletter_Email-statistics-grid-ribbon-implementation'.

jamesallenuk committed:
PRESIDECMS-3133 / PRESIDECMS-3134 Added email template stats to Email and Newsletters admin and enabled data export

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/pixl8/Preside-CMS?pullRequest=1640) <br/>All committers have signed the CLA.

DominicWatson commented on system/handlers/admin/emailCenter/CustomTemplates.cfc:
```diff
@@ -108,7 +111,7 @@ component extends="preside.system.base.AdminHandler" {
 		prc.filterObject = emailRecipientTypeService.getFilterObjectForRecipientType( prc.template.recipient_type );
 		prc.canPreviewWithRecipient = Len( Trim( prc.filterObject ) );
-		if ( prc.canPreviewWithRecipient && Len( Trim( previewRecipient ) ) ){
+				if ( prc.canPreviewWithRecipient && Len( Trim( previewRecipient ) ) ){
```
indentation change here should not be here. AI code editor messing??

DominicWatson commented on system/handlers/admin/emailCenter/CustomTemplates.cfc:
```diff
@@ -202,7 +205,7 @@ component extends="preside.system.base.AdminHandler" {
 
 		prc.pageTitle    = translateResource( uri="cms:emailcenter.customTemplates.edit.page.title", data=[ prc.record.name ] );
-		prc.pageSubtitle = translateResource( uri="cms:emailcenter.customTemplates.edit.page.subtitle", data=[ prc.record.name ] );
+				prc.pageSubtitle = translateResource( uri="cms:emailcenter.customTemplates.edit.page.subtitle", data=[ prc.record.name ] );
```
indentation change here should not be here. AI code editor messing??

DominicWatson commented on system/handlers/admin/emailCenter/CustomTemplates.cfc:
```diff
@@ -527,11 +530,12 @@ component extends="preside.system.base.AdminHandler" {
+				, actionsView             = "admin.emailCenter/customTemplates._gridActions"
+				, filter                  = { "email_template.is_system_email" = false }
+				, draftsEnabled           = true
+				, includeAllFormulaFields = true
```
I think this _shouldn't_ be needed as we explicitly specify the fields we want. Unless we're using custom renderers where we fetch some formula fields. Better to use hiddenGridFields to specify particular fields we want if so - otherwise we could end up fetching and calculating expensive formula fields in the future that we don't need.

jamesallenuk committed:
PRESIDECMS-3133 Removed include all formula fields argument and cleaned up indentation

jamesallenuk committed:
PRESIDECMS-3133 Fixed incorrect stats calculations

By filtering the sendCount to 1 year, the rest of the stats were incorrect as they are not filtered. Returned to filtering sendCount over all time but including a sendCountFiltered to allow the display of 1 year worth of emails sent.

jamesallenuk committed:
PRESIDECMS-3133 Fixed missing % values on export

jamesallenuk committed:
PRESIDECMS-3133 Made new stat fields sortable in datamanager list

jamesallenuk committed:
PRESIDECMS-3133 Fix email template stats inconsistency between grid and detail views

- Update EmailOpenRate and EmailClickRate renderers to use EmailStatsService
- Ensures grid calculations match performance tab by using same data source
- Both views now consistently use email_template_stats table instead of mixed sources
- Resolves discrepancies caused by stats migration timing differences between environments

jamesallenuk committed:
PRESIDECMS-3133 Improved database send_count efficiency call
